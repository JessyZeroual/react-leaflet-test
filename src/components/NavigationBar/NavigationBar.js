import React from "react";
import CityList from "../CityList/CityList";
import FilterCities from "../FilterCities/FilterCities";
import {
  StyledNavigationBar,
  StyledHeaderNavigationBar,
  StyledMainNavigationBar,
  // StyledFooterNavigationBar,
} from "./NavigationBar.styled";

const NavigationBar = () => (
  <StyledNavigationBar>
    <StyledHeaderNavigationBar>
      <FilterCities />
    </StyledHeaderNavigationBar>
    <StyledMainNavigationBar id="mainSideBarID">
      <CityList />
    </StyledMainNavigationBar>
    {/* <StyledFooterNavigationBar>footer</StyledFooterNavigationBar> */}
  </StyledNavigationBar>
);

export default NavigationBar;
