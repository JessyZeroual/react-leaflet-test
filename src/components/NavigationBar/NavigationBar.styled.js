import styled from "styled-components";
import {
  NavigationBarWidth,
  secondaryColor,
} from "../../constant/style-constants";

export const StyledNavigationBar = styled.aside`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: ${NavigationBarWidth}px;
  padding: 10px;
  background: #f4f4f6;
  border-top-left-radius: 15px;
  border-top-right-radius: 15px;
`;

export const StyledHeaderNavigationBar = styled.div`
  position: fixed;
  width: ${NavigationBarWidth}px;
  height: 30px;
`;

export const StyledMainNavigationBar = styled.div`
  margin-top: 50px;
  overflow: auto;
  ::-webkit-scrollbar {
    background-color: transparent;
    border-radius: 20px;
    width: 6px;
  }
  ::-webkit-scrollbar-thumb {
    background-color: #c0c0c0;
    border-radius: 20px;
  }
  &:hover {
    ::-webkit-scrollbar-thumb {
      background-color: ${secondaryColor};
    }
  }
`;
export const StyledFooterNavigationBar = styled.div`
  height: 60px;
`;
