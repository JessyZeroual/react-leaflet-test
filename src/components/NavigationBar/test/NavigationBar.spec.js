import React from "react";
import TestRenderer from "react-test-renderer";
import NavigationBar from "../NavigationBar";
import { StyledMainNavigationBar } from "../NavigationBar.styled";

jest.mock("../../CityList/CityList", () => () => <div></div>);
jest.mock("../../FilterCities/FilterCities", () => () => <div></div>);

describe("NavigationBar", () => {
  let renderer;

  beforeEach(() => {
    renderer = TestRenderer.create(<NavigationBar />);
  });

  describe("When NavigationBar is mounted", () => {
    it("renders correctly", () => {
      expect(renderer).toMatchSnapshot();
    });

    it("has a StyledMainNavigationBar component with an id named mainSideBarID", () => {
      expect(
        renderer.root.findByType(StyledMainNavigationBar).props.id
      ).toEqual("mainSideBarID");
    });
  });
});
