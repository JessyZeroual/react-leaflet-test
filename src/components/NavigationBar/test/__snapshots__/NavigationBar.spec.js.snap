// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar When NavigationBar is mounted renders correctly 1`] = `
<aside
  className="sc-AxjAm jNNAvW"
>
  <div
    className="sc-AxirZ iPIdxu"
  >
    <div />
  </div>
  <div
    className="sc-AxiKw FVqab"
    id="mainSideBarID"
  >
    <div />
  </div>
</aside>
`;
